File: LeetCode Problems/201 to 300/209. Minimum Size Subarray Sum/209. Minimum Size Subarray Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int sum = 0, l = 0, length = 1000000007;
        for (int r = 0; r < nums.length; r++) {
            sum += nums[r];
            while (sum >= target) {
                length = Math.min(length, r - l + 1);
                sum -= nums[l];
                l++;
            }
        }
        return length == 1000000007 ? 0 : length;
    }
}

File: LeetCode Problems/201 to 300/209. Minimum Size Subarray Sum/209. Minimum Size Subarray Sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int> &nums) {
        int sum = 0, length = INT_MAX, l = 0;
        for (int r = 0; r < nums.size(); r++) {
            sum += nums[r];
            while (sum >= target) {
                length = min(length, r - l + 1);
                sum -= nums[l];
                l++;
            }
        }
        return length == INT_MAX ? 0 : length;
    }
};